src/event/event.controller.ts

```diff
@@ -1,7 +1,36 @@
-import { Controller } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Post,
+  UploadedFiles,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
 import { EventService } from './event.service';
+import { AuthGuard } from '../auth/auth.guard';
+import { CreateEventDto } from './create-event.dto';
+import { FileFieldsInterceptor } from '@nestjs/platform-express';
 
-@Controller('event')
+@Controller('/events')
 export class EventController {
   constructor(private readonly eventService: EventService) {}
+
+  @Post()
+  @UseGuards(AuthGuard)
+  @UseInterceptors(FileFieldsInterceptor([{ name: 'poster_images' }]))
```

Comment:
Should validate that the uploaded files for `[poster_images]` in valid image formats else throw an error